### DISCUSSION TITLE
Session Menu - 32 Max Sessions

### DISCUSSION THREAD

**Discussion by uforce** [neutrinolabs/xrdp#2912] [Q&A] [answered]
Hello,

Is there anyway to increase the amount of sessions in the drop down past 32?

Are you can see after 32 connections the menu just stops and there's no way to scroll.

![image](https://github.com/neutrinolabs/xrdp/assets/19957863/5fabb43b-b2db-4552-8644-a8279cf014d0)

**Comment by matt335672** [comment]
I'm away from my test environment at the moment, so I can't reproduce this right now.

As far as I can see from the source, the list should only be limited by your screen size. There's no provision for a scroll bar on it.

A couple of questions for you:-
1) How big is your xrdp.ini file?
2) If you select the combo without dropping the list down you can use the arrow keys to select items. If you do that, are other list items accessible?

**Reply by uforce** [reply]
Hey Matt,

I have 38 connections - could grow more.

If I select the box without the drop down I can use the arrow keys to go up and down.

Is there no way to have a scroll bar on this menu?

**Reply by matt335672** [reply]
A scroll bar would be somewhat challenging. The xrdp login screen uses the raw RDP protocol to draw things, so we don't have a widget set as such available to us.

By 'how big is your xrdp.ini file' I wondered about the size in bytes. This file is currently limited to 32K in size I believe, looking at the code.

Is there space under the menu for additional entries, or is that it for screen space?

**Reply by uforce** [reply]
Thanks for the info...

The xrdp.ini file is at 18K right now so there is still some room left.

Yeah the screen shot I posted is the end of available screen space on a 1080p monitor. Obviously if I use this on one of my 2K/4K monitors I can see more sessions.

**Reply by matt335672** [reply, marked as answer]
At the moment I can't suggest any more than using the arrow key trick I'm afraid.
